src/utils/abi_decoder.ts

```diff
@@ -10,40 +10,43 @@ export class AbiDecoder {
     constructor(abiArrays: Web3.AbiDefinition[][]) {
         _.map(abiArrays, this.addABI.bind(this));
     }
-    public decodeLog(logItem: Web3.LogEntry): DecodedArgs|undefined {
-        const methodId = logItem.topics[0];
+    public tryToDecodeLogOrNoOp(log: Web3.LogEntry): LogWithDecodedArgs|RawLog {
```

Comment:
Add comment above this method explaining the scenarios where it will be unable to decode the logs.